### DISCUSSION TITLE
CLEAR ALL not reseting dropdowns correctly

### DISCUSSION THREAD

**Discussion by vdiazs** [apache/superset#34150] [Ideas]
#### How to reproduce the bug

1. Click on "CLEAR ALL"

### Expected results

That the dropdowns show the default value

### Actual results

It leaves the field empty and "applies filters" showing the wrong data

#### Screenshots

[If applicable, add screenshots to help explain your problem.
](https://github.com/apache/superset/assets/58532225/3669f8fa-4ed2-427b-9d22-bd184631980c)

### Environment

(please complete the following information):

- browser type and version: Chrome (Version 117.0.5938.92)
- superset version: 2.10
- python version: docker image ( apache/superset:2.1.0)

- node.js version: `node -v`
- any feature flags active:

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [X] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [X] I have reproduced the issue with at least the latest released version of superset.
- [X] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

Also, I don't think that that clear filters should trigger a reload of the data.  I would rather see that the apply filters enables the "APPLY  FILTERS" button.  The reason being that when you have multiple filters  you just want to clear them and restart the process to make a new selection, not necesarily to reload the original data.

**Comment by marcelorafaelfeil** [comment]
Hi,

I would like to work on this item. Could you assign to me?

**Comment by marcelorafaelfeil** [comment]
Thanks

**Comment by rusackas** [comment]
Hi @marcelorafaelfeil if you're still interested in working on this, you're quite welcome to.  I agree with you that the default would be the correct behavior, especially considering that these default values might be there to prevent performance/query time issues. 

That said, I must admit there's a little part of me that pedantically wants to argue that "clear all" does exactly what it says, and "reset all" would be more fitting to restore default values.

**Comment by rusackas** [comment]
CC @kasiazjc @yousoph in case there's an argument to be made against the proposed change (I'm not sure if this ever came up in a design discussion at some point).

**Comment by zwickera** [comment]
Could there be 2 options, one to "Clear All" and one to "Reset All"?  Having a "Reset All" to set the defaults would be a good improvement to superset.  It would be good to have both as options.

**Comment by kasiazjc** [comment]
> Could there be 2 options, one to "Clear All" and one to "Reset All"? Having a "Reset All" to set the defaults would be a good improvement to superset. It would be good to have both as options.

I think that makes sense, although with both options it can get too busy and confusing, so I wouldn't put them in one place. 

Maybe we could leave "clear all" below the button - it would clear all, even default values and each field would be empty, no filters applied. 
In the settings/gear icon we could have "restore default values" and that would bring back all of the defaults. 

Thoughts @rusackas @yousoph ?

**Comment by rusackas** [comment]
I'm not sure if anyone is still interested in taking this on, but I'm happy with either the Gear approach, or this style of button, with "Clear all" and "Clear and Reset"

<img width="133" alt="Image" src="https://github.com/user-attachments/assets/6ca9d65f-5a54-4f90-8ced-c2531425d245" />

Anyone planning to tackle this? I'm not sure if we should leave it as a bug or move it to an Ideas discussion.

**Comment by msyavuz** [comment]
Moving this to discussions since it seems like a feature request for a seperate `Reset all` button.
